src/test/scala/ir/IRTest.scala

```diff
@@ -0,0 +1,383 @@
+      call("called", None)
+    ).resolve(p)
+
+    assert(b3.calls.toSet == Set(p.procs("called")))
+    val oldb = blocks("lmain2")
+    p.mainProcedure.replaceBlock(blocks("lmain2"), b3)
+
+    assert(p.mainProcedure.calls.toSet == Set(p.procs("called")))
+    assert(p.mainProcedure.calls.forall(_.callers().exists(_ == p.mainProcedure)))
+    assert(!oldb.hasParent)
+    assert(oldb.incomingJumps.isEmpty)
+    assert(!blocks("lmain").jump.asInstanceOf[GoTo].targets.contains(oldb))
+    assert(called.incomingCalls().toSet == Set(b3.jump))
+    assert(called.incomingCalls().map(_.parent.parent).toSet == called.callers().toSet)
+    val olds = blocks.size
+    p.mainProcedure.replaceBlock(b3, b3)
+    assert(called.incomingCalls().toSet == Set(b3.jump))
+    assert(olds == blocks.size)
+    p.mainProcedure.addBlocks(block("test", ret).resolve(p))
+    assert(olds != blocks.size)
+
+    p.mainProcedure.replaceBlocks(Set(block("test", ret).resolve(p)))
+    assert(blocks.count(_._2.parent.name == "main") == 1)
+
+  }
+
+  test("clearblocks") {
+    val p = prog(
+        proc("main",
+          block("l_main",
+            LocalAssign(R0, bv64(10)),
+            LocalAssign(R1, bv64(10)),
+            goto("returntarget")
+          ),
+          block("returntarget",
+            ret
+          )
+        ),
+      )
+
+    assert(p.blocks.size > 1)
+    assert(p.procs("main").entryBlock.isDefined)
+    p.procs("main").returnBlock = block("retb", ret).resolve(p)
+    assert(p.procs("main").returnBlock.isDefined)
+    p.procs("main").clearBlocks()
+    assert(p.blocks.isEmpty)
+    assert(p.procs("main").entryBlock.isEmpty)
+    assert(p.procs("main").returnBlock.isEmpty)
+  }
+
+  test("interproc aftercall") {
+
+    val p = prog(
+      proc("p1",
+        block("b1",
+          LocalAssign(R0, bv64(10)),
+          ret
+        )
+      ),
+      proc("main",
+        block("l_main",
+          LocalAssign(R0, bv64(10)),
+          call("p1", Some("returntarget"))
+        ),
+        block("returntarget",
+          ret
+        )
+      ),
+    )
+    val returnUnifier = ConvertToSingleProcedureReturn()
+    returnUnifier.visitProgram(p)
+
+    val next = InterProcIRCursor.succ(p.blocks("l_main").jump)
+    val prev = InterProcIRCursor.pred(p.blocks("returntarget"))
+
+    assert(prev.size == 1 && prev.collect {
+      case c : GoTo => (c.parent == p.blocks("l_main")) && c.isAfterCall
+    }.contains(true))
+
+    assert(next.toSet == Set(p.procs("p1"), p.blocks("l_main").fallthrough.get))
+
+    val prevB : Block = (p.blocks("l_main").jump match
+      case c : IndirectCall  => c.returnTarget
+      case c : DirectCall => c.returnTarget
+    ).get
+
+    assert(prevB.isAfterCall)
+    assert(InterProcIRCursor.pred(prevB).size == 1)
+    assert(InterProcIRCursor.pred(prevB).toSet == Set(p.blocks("l_main").fallthrough.get))
+    assert(InterProcBlockIRCursor.pred(prevB).toSet == Set(p.blocks("l_main"), p.procs("p1").returnBlock.get))
```

Comment:
These .toSets (and some others) are redundant

Reply:
They're defensive in case the IR gets refactored to not be a set anymore

Reply:
wouldn't it make more sense to use .head (and not convert the rhs to a set) given that the previous line is an assertion that there is only 1 predecessor